### DISCUSSION TITLE
Need help with my app, I'll pay for solution because it's complex

### DISCUSSION THREAD

**Discussion by itstermoh** [community/community#68051] [Programming Help]
I recently bought an ir blaster for my android that connects via otg port. I'm currently making an app to control every
kind of device of all brands. I installed the app made for this ir and works perfectly, so I wanted to decompile it and get some code to implement the ir in my app. I tried to recreate many functions from decompiled code and made the permission request but it didn't work

**Comment by OSEight** [comment]
Knowing that Android applications are made using HTML, I can assume that if you use a computer to find the file off your Android device, you can save the file and deconvert it with a tool such as JADX. To fully access your Android device, you can search online about it with the format of "How to root the [Device Type] " replacing [Device Type] with, well, your device type. 
  I have assumed you have had Andriod, so on an iPhone would need to be homebrewed to do this. A simple way of doing this is by going to the [3uTools official site](3u.com) and using that. Thank you for reading! Hope it helped.
